internal/evaluator/conftest_evaluator.go

```diff
@@ -65,6 +65,61 @@ func (c CheckResults) ToConftestResults() []output.CheckResult {
 	return results
 }
 
+// trim removes all failure, warning, success or skipped results that depend on
+// a result reported as failure, warning or skipped. Dependencies are declared
+// by setting the metadata via metadataDependsOn.
+func (c *CheckResults) trim() {
+	// holds codes for all failures, warnings or skipped rules, as a map to ease
+	// the lookup, any rule that depends on a reported code will be removed from
+	// the results
+	reported := map[string]bool{}
+
+	for _, checks := range *c {
+		for _, results := range [][]output.Result{checks.Failures, checks.Warnings, checks.Skipped} {
```

Comment:
I forget why this is a slice of slices. Is the first dimension meant to represent source groups? I wonder if we should take care to ensure one source group doesn't affect another one. I _think_ that would be the right behavior.

Reply:
This is just a shorter way of writing the loop, so I don't need to loop over checks.Failures, checks.Warnings and checks.Skipped in three loops...

IIRC we have an Evaluator per source group, but we should test (created [HACBS-2257](https://issues.redhat.com/browse/HACBS-2257))